meta/recipes-extended/libnsl/libnsl2_git.bb

```diff
@@ -21,4 +21,11 @@ S = "${WORKDIR}/git"
 
 inherit autotools pkgconfig gettext
 
+# on eSDK population, nativesdk-nsl2 and nativesdk-glibc provide yppasswd.x and
+# yppasswd.h which causes conflict on do_recipe_prepare_sysroot()
+do_install_append_libc-glibc_class-nativesdk() {
+    rm -f ${D}${includedir}/rpcsvc/yppasswd.x
+    rm -f ${D}${includedir}/rpcsvc/yppasswd.h
```

Comment:
are these files different between glibc and libnsl2 ? if so what are differences?
I think it should be provided by libnsl2 which means it should 
be deleted from glibc recipe

Reply:
There is no difference in yppasswd.x but some API in glibc's yppasswd.h are declared under __STDC__ and __cplusplus macros as below
`
#if defined(__STDC__) || defined(__cplusplus)
#define YPPASSWDPROC_UPDATE 1
extern  int * yppasswdproc_update_1(yppasswd *, CLIENT *);
extern  int * yppasswdproc_update_1_svc(yppasswd *, struct svc_req *);
extern int yppasswdprog_1_freeresult (SVCXPRT *, xdrproc_t, caddr_t);
#else /* K&R C */
#define YPPASSWDPROC_UPDATE 1
extern  int * yppasswdproc_update_1();
extern  int * yppasswdproc_update_1_svc();
extern int yppasswdprog_1_freeresult ();
#endif /* K&R C */

/* the xdr functions */
#if defined(__STDC__) || defined(__cplusplus)
extern  bool_t xdr_passwd (XDR *, passwd*);
extern  bool_t xdr_yppasswd (XDR *, yppasswd*);
#else /* K&R C */
extern bool_t xdr_passwd ();
extern bool_t xdr_yppasswd ();
#endif /* K&R C */
`

Reply:
another observation that the nativesdk-glibc in yoe build doesn't populate /usr/include/rpcsvc directory so that no error faced. The difference observed is yoe generates esdk toolchain installer version is 3.3.0-beta whereas from usual oe-core dunfell build is 2.0. Is that a reason for the error?

Reply:
an equivalent fix found at https://git.yoctoproject.org/cgit/cgit.cgi/meta-arm/tree/meta-arm-toolchain/recipes-devtools/external-arm-toolchain/external-arm-toolchain.bb?h=dunfell#n207